src/AltinnCore/Designer/Controllers/ModelController.cs

```diff
@@ -51,45 +56,45 @@ public ActionResult Index(string org, string service)
         /// <param name="org">The Organization code for the service owner</param>
         /// <param name="service">The service code for the current service</param>
         /// <param name="thefile">The main XSD</param>
-        /// <param name="secondaryFiles">Secondary xsd</param>
         /// <returns>Return JSON of the generated model</returns>
         [HttpPost]
-        public ActionResult Upload(string org, string service, IFormFile thefile, IEnumerable<IFormFile> secondaryFiles)
+        public ActionResult Upload(string org, string service, IFormFile thefile)
         {
             if (thefile == null)
             {
                 throw new ApplicationException("Cannot upload empty file");
             }
 
-            XDocument mainXsd = null;
-            var secondaryXsds = new Dictionary<string, XDocument>();
-
             string mainFileName = ContentDispositionHeaderValue.Parse(new StringSegment(thefile.ContentDisposition)).FileName.ToString();
 
-            XmlReaderSettings settings = new XmlReaderSettings();
-            settings.IgnoreWhitespace = true;
+            MemoryStream xsdMemoryStream = new MemoryStream();
+            thefile.OpenReadStream().CopyTo(xsdMemoryStream);
+            xsdMemoryStream.Position = 0;
+            XmlReader reader = XmlReader.Create(xsdMemoryStream, new XmlReaderSettings { IgnoreWhitespace = true });
 
-            using (var reader = XmlReader.Create(thefile.OpenReadStream(), settings))
-            {
-                mainXsd = XDocument.Load(reader, LoadOptions.None);
-            }
+            XDocument mainXsd = XDocument.Load(reader, LoadOptions.None);
 
-            secondaryXsds.Clear();
+            ServiceMetadata serviceMetadata = null;
 
-            if (secondaryFiles != null)
```

Comment:
Are we no longer going to support secondary files? Maybe the btn can be removed as well if so

Reply:
We never actually did support secondaryFiles. It was ignored by the controller. Therefore we have taken it away.